static/js/components/widgets/ObjectField.tsx

```diff
@@ -42,13 +45,15 @@ export default function ObjectField(props: Props): JSX.Element {
       </div>
       {collapsed ? null : (
         <div className="object-sub-fields">
-          {field.fields?.map((field: ConfigField) => (
-            <SiteContentField
-              field={field}
-              key={field.name}
-              contentContext={contentContext}
-            />
-          ))}
+          {field.fields
+            ?.filter(innerField => fieldIsVisible(innerField, values))
```

Comment:
I think we can get rid of the `?` here because as long as we're dealing with an object field is should always have fields defined